### PR TITLE
Fastify swagger implementation

### PR DESCRIPTION
**Pull request by panbhatt**
https://github.com/Ethan-Arrowood/matterhorn/issues/22 

Fixed with linting Rules.

### CODE REVIEW

**Comment by Ethan-Arrowood** [comment]
Looks like `npm run test` is failing too. I pulled your changes locally and it looks like the schema you're passing is invalid. Try fixing that so `npm run test` passes

**Comment by Ethan-Arrowood** [comment]
This has something to do with the TypeScript types. I believe fastify-swagger is adding additional properties to fastify's RouteSchema interface as defined in fastify.d.ts. I believe you'll need to open a PR to fastify-swagger that extends the RouteSchema interface to include the properties used by swagger.

**Comment by panbhatt** [comment]
Just corrected the schema and wrote the unit test case and now npm test is passing.

**Comment by Ethan-Arrowood** [comment]
I don't think removing those properties is the correct change. You'll need to extend fastify's RouteSchema interface in `fastify-swagger` [index.d.ts](https://github.com/fastify/fastify-swagger/blob/master/index.d.ts) file. You'll want to add 
```js
interface RouteSchema {
    description: String,
    tags: String[],
    summary: String,
    security: any[]
  }
``` 
to this module block https://github.com/fastify/fastify-swagger/blob/55cba82f6ec9823a9d6f50accd5ce64ca9b85d1b/index.d.ts#L45

**Comment by panbhatt** [comment]
I am little confused via This. Your suggestion seems correct. However RouteSchema is already defined in Fastify.d.ts. I think we need to add it there, rather then in fastify-swagger. In case I extends it in fastify-swagger.ts with 
` interface RouteSchema extends fastify.RouteSchema{
    description: String,
    tags: String[],
    summary: String,
    security: any[]
  }`
how will that make change to Fastify Schema. i.e. when we are defining the methods, it will link itself to RouteSchema in fastify.d.ts. Please help.

**Comment by Ethan-Arrowood** [comment]
In TypeScript world when you want to expand upon an already existing type (as in modify) but you cannot directly change the type source then you simply redeclare the module using the same name and TypeScript compiler will merge the two together. `fastify-swagger` is already doing this:
```js
declare module 'fastify' {
  interface FastifyInstance<
    HttpServer = http.Server,
    HttpRequest = http.IncomingMessage,
    HttpResponse = http.ServerResponse
  > {
    swagger: (
      opts?: {
        yaml?: boolean;
      }
    ) => SwaggerSchema.Spec;
  }
}
```
(copied and pasted from index.d.ts)
In this snipped, fastify-swagger is expanding the FastifyInstance interface to include the `swagger` property. 

You'll do the same thing when expanding `RouteSchema`

File: tests/index.test.ts [hunk 1]
```diff
@@ -23,3 +23,15 @@ test('should return twitter handle', async () => {
     expect(error).toBeFalsy()
   }
 })
+
+
+test('should return swager page on /documentation', async () => {
+  const handle = '@ArrowoodTech'
```
Comment (Ethan-Arrowood) [inline tests/index.test.ts]: Remove unused variable

File: src/plugins/index.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          in: 'header'
+        }
+      }
+    },
+    exposeRoute: true
+  })
+
+} 
+
+/*
```
Comment (Ethan-Arrowood) [inline src/plugins/index.ts]: Remove this comment and the empty function `configureOtherThings`

**Comment by panbhatt** [comment]
Just raise the PULL Request, once it is merged, I will put the GIT URL in package.json and will update the code. https://github.com/fastify/fastify-swagger/pull/157

**Comment by Ethan-Arrowood** [comment]
@panbhatt fastify-swagger was update so next you'll need to update it in your package.json. Then add back in the correct parameters.

**Comment by Ethan-Arrowood** [comment]
Hi @panbhatt i'm back from my trip so I'll spend some time on your PR today. Expect some feedback soon!

File: src/plugins/index.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import * as FastifySwagger from 'fastify-swagger' 
+import * as Fastify from 'fastify'
```
Comment (Ethan-Arrowood) [inline src/plugins/index.ts]: I suggest you change this line to:
```suggestion
/* eslint-disable-next-line no-unused-vars */
import { FastifyInstance } from 'fastify'
```

File: src/plugins/index.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+export  function configureSwagger(fastifyInstance: Fastify.FastifyInstance ) {
```
Comment (Ethan-Arrowood) [inline src/plugins/index.ts]: And change this one to:
```suggestion
export function configureSwagger(fastifyInstance: FastifyInstance) {
```

File: src/server.ts [hunk 5]
```diff
@@ -1,10 +1,13 @@
 import * as Fastify from 'fastify'
+import { configureSwagger } from './plugins'
```
Comment (Ethan-Arrowood) [inline src/server.ts]: I'd change the naming here so its a bit more declarative
```suggestion
import { configureSwaggerPlugin } from './plugins'
```

File: src/server.ts [hunk 6]
```diff
@@ -1,10 +1,13 @@
...
 
 export default function createServer(opts?: Fastify.ServerOptions) {
   const fastify = Fastify(opts)
 
+  configureSwagger(fastify)
```
Comment (Ethan-Arrowood) [inline src/server.ts]:
```suggestion
  configureSwaggerPlugin(fastify) 
```

File: src/server.ts [hunk 7]
```diff
@@ -1,10 +1,13 @@
...
+import { configureSwagger } from './plugins'
 
 export default function createServer(opts?: Fastify.ServerOptions) {
   const fastify = Fastify(opts)
 
+  configureSwagger(fastify) 
+
   /* eslint-disable-next-line no-unused-vars */
-  fastify.get('/', async (request, reply) => {
+  fastify.get('/', async (request , reply) => {
```
Comment (Ethan-Arrowood) [inline src/server.ts]: Remove the whitespace added here

File: src/server.ts [hunk 8]
```diff
@@ -13,5 +16,61 @@ export default function createServer(opts?: Fastify.ServerOptions) {
     return { twitterHandle: request.query.handle }
   })
 
+
+  // Fastify Swagger Test URI demonstration i.e. sample Path.
```
Comment (Ethan-Arrowood) [inline src/server.ts]: Move this `.post` block to the configureSwaggerPlugin function

File: src/server.ts (hunk 5 above)
Comment (panbhatt) [reply src/server.ts]: Done.

File: src/server.ts (hunk 6 above)
Comment (panbhatt) [reply src/server.ts]: Done.

File: src/server.ts (hunk 7 above)
Comment (panbhatt) [reply src/server.ts]: Done.

File: src/server.ts (hunk 8 above)
Comment (panbhatt) [reply src/server.ts]: Done.

**Comment by panbhatt** [comment]
Hey, All suggested changes are done. however CI test is failing, please can you have a look at let me know what needs to be done there by me (if anything).

**Comment by Ethan-Arrowood** [comment]
Awesome work. The issue is not your fault, something on Azure's side. I'll figure it out and get back to you.

**Comment by Ethan-Arrowood** [comment]
@panbhatt can you give me push access to your branch? I'd like to try and update the package-lock which I believe may be apart of the issue

**Comment by panbhatt** [comment]
Just did.

**Comment by Ethan-Arrowood** [comment]
Thank you, I'll update you when things change

**Comment by panbhatt** [comment]
Let me know, if you want me to work on some other feature. I will try to make it as solid as i can.

**Comment by Ethan-Arrowood** [comment]
I'm having issues pushing to your branch still

**Comment by Ethan-Arrowood** [comment]
Check branch permissions maybe

**Comment by Ethan-Arrowood** [comment]
Disregard previous comments. I was pushing wrong (forgot my local branch had different name than your remote one). Checking ci now

**Comment by panbhatt** [comment]
oh Great. i thought i did some mistake. The page says awaiting your response. 
https://github.com/panbhatt/matterhorn/invitations

**Comment by Ethan-Arrowood** [comment]
Yeah you're all good, I'm still fighting the CI. Consider this complete though. Once I figure out why the CI is failing I will push and merge your PR. Great work 👍

**Comment by Ethan-Arrowood** [comment]
Hey @panbhatt so I've spent a while trying to figure this out. Check out my issue: https://github.com/fastify/fastify-swagger/issues/162 if you want some more details. 

For now I'm going to mark this PR as blocked. When I get some help on resolving it i'll merge it into master. Thank you again for your contribution. If you'd like to try and solve this yourself, I encourage you to attempt it! Maybe you'll see something I don't.

**Comment by Ethan-Arrowood** [comment]
🎉 figure it out! merging now 😄 if you're looking for more ways to contribute I think adding more fastify plugins like this will be a great way to get involved.

**Comment by panbhatt** [comment]
That's Great. Let me know wherever u think i can add like writing CLI  etc, i will jump in to make it a better product and improve myself.

**Pull request merged**